pytryfi/fiPet.py

```diff
@@ -42,11 +42,31 @@ def setPetDetailsJSON(self, petJSON):
             self._device.setDeviceDetailsJSON(petJSON['device'])
         except Exception as e:
             capture_exception(e)
+        self._connectedTo = self.setConnectedTo(petJSON['device']['lastConnectionState'])
 
     def __str__(self):
-        return f"Last Updated - {self.lastUpdated} - Pet ID: {self.petId} Name: {self.name} Is Lost: {self.isLost} From: {self.homeCityState} ActivityType: {self.activityType} Located: {self.currLatitude},{self.currLongitude} Last Updated: {self.currStartTime}\n \
+        return f"Last Updated - {self.lastUpdated} - Pet ID: {self.petId} Name: {self.name} Connected To: {self.connectedTo} Is Lost: {self.isLost} From: {self.homeCityState} ActivityType: {self.activityType} Located: {self.currLatitude},{self.currLongitude} Last Updated: {self.currStartTime}\n \
             using Device/Collar: {self._device}"
     
+    # set the connected to friendly string
+    def setConnectedTo(self, connectedToJSON):
+        connectedToString = ""
+        try:
+            typename = connectedToJSON['__typename']
+            if typename == 'ConnectedToUser':
+                connectedToString = connectedToJSON['user']['firstName'] + " " + connectedToJSON['user']['lastName']
```

Comment:
It looks like you're stringifying the state down in a library. Is there any way to leave it structured and make the calling app format it? I'd use this feature but generally a library like this shouldn't force a particular string format pattern.